### PR TITLE
Python 3.12 changes

### LINKED ISSUE #41: Python 3.12 builds fail because `imp` has been deprecated and removed
Title, https://docs.python.org/3.11/library/imp.html

### PR DESCRIPTION
**Pull request by dholth**
Changes needed for Python 3.12, distutils removal, pkg_resources removal.

Fix #41

### CODE REVIEW

File: enscons/__init__.py [hunk 1]
```diff
@@ -187,7 +183,7 @@ def egg_info_builder(target, source, env):
     """
     Minimum egg_info. To be used only by pip to get dependencies.
     """
-    metadata = env["PACKAGE_METADATA"]
+    env["PACKAGE_METADATA"]
```
Comment (richardkiss) [inline enscons/__init__.py]: This deserves a comment since it doesn't seem like it does anything (and in fact, I'm not really sure what it's purpose is).

**Comment by richardkiss** [comment]
Does this remove support for python <= 3.10? If so, you might consider adding

```
[project]
...
requires-python = ">=3.11"
```

**Comment by dholth** [comment]
> Does this remove support for python <= 3.10? If so, you might consider adding
> 
> ```
> [project]
> ...
> requires-python = ">=3.11"
> ```

Still works in [Python 3.9, 3.10 at least](https://github.com/dholth/enscons/pull/44/files#diff-b1b82537432942b954fa8846c9e80709d0d919428ed1e1692d7c0bc4bc8271ecR10) as checked by this workflow by building it with pypy 3.9, 3.10

File: .github/workflows/upload-pypi-source.yml [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-on: [push, pull_request]
+on: [pull_request]
 
 jobs:
   build:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.8", "3.9", "3.10", pypy3.9]
+        python-version: ["3.11", "3.12", "pypy3.9", "pypy3.10"]
```
Comment (richardkiss) [inline .github/workflows/upload-pypi-source.yml]: Probably want to add back in 3.8, 3.9, 3.10 here.

**Pull request merged**